qml/PingSlider.qml

```diff
@@ -24,6 +25,73 @@ RowLayout {
         Layout.fillWidth: true
         stepSize: 1
         wheelEnabled: true
+
+        background: Rectangle {
+            x: sliderControl.leftPadding
+            y: sliderControl.topPadding + sliderControl.availableHeight/2 - height/2
+            implicitWidth: 200
+            implicitHeight: 1
+            width: sliderControl.availableWidth
+            height: implicitHeight
+            color: Material.accent
+
+            Repeater {
+                id: internalTicks
+                model : sliderControl.to - sliderControl.from
+
+                delegate: Rectangle {
+                    id: tick
+                    width: 1
+                    visible: {
+                        var count = sliderControl.to - sliderControl.from;
+                        if (count < 100) {
+                            return true;
+                        }
+                        else if (count < 250) {
+                            return (index + 1) % 5 == 0
+                        } else {
+                            return (index + 1) % 20 == 0
+                        }
+                    }
+
+                    height: parent.implicitHeight * (index % 10 == 0 ?  6 : 3)
+
+                    x: {
+                        var count = sliderControl.to - sliderControl.from;
+                        return index * sliderControl.availableWidth / count
+                    }
+
+                    y: parent.implicitHeight
```

Comment:
this should be 0 or not exist, otherwise you'll need to use parentImplicitHeight in math conditions or operations to get the real size.